jumpscale/sals/nettools/__init__.py

```diff
@@ -0,0 +1,457 @@
+
+
+def get_nics():
+    """Get Nics on this machine
+
+    Returns:
+        list: list of all availabe nics
+    """
+    return [nic["name"] for nic in get_network_info()]
+
+
+def get_nic_type(interface):
+    """Get Nic Type on a certain interface
+
+    Args:
+        interface (str): interface name
+
+    Raises:
+        j.exceptions.Runtime: if ethtool not installed on the system
+        j.exceptions.Value: if interface given is invalid
+
+    Returns:
+        str: type of the interface
+    """
+    output = ""
+    if j.data.platform.is_linux():
+        if j.sals.fs.exists(f"/sys/class/net/{interface}"):
+            output = j.sals.fs.read_file(f"/sys/class/net/{interface}/type")
+        if output.strip() == "32":
+            return "INFINIBAND"
+        else:
+            if j.sals.fs.exists("/proc/net/vlan/%s" % (interface)):
+                return "VLAN"
+            exitcode, _, _ = j.core.executors.run_local("which ethtool", hide=True, warn=True)
+            if exitcode != 0:
+                raise j.exceptions.Runtime("Ethtool is not installed on this system!")
+            exitcode, output, _ = j.core.executors.run_local(f"ethtool -i {interface}", hide=True, warn=True)
+            if exitcode != 0:
+                return "VIRTUAL"
+            match = re.search(r"^driver:\s+(?P<driver>\w+)\s*$", output, re.MULTILINE)
+            if match and match.group("driver") == "tun":
+                return "VIRTUAL"
+            if match and match.group("driver") == "bridge":
+                return "VLAN"
+            return "ETHERNET_GB"
+
+    elif j.data.platform.is_osx():
+        command = f"ifconfig {interface}"
+        exitcode, output, err = j.core.executors.run_local(command, hide=True, warn=True)
+        if exitcode != 0:
+            # temporary plumb the interface to lookup its mac
+            print(
+                "Interface %s is down. Temporarily plumbing it to be able to lookup its nic type" % interface
+            )
+            j.core.executors.run_local(f"{command} plumb", hide=True)
+            exitcode, output, err = j.core.executors.run_local(command, hide=True)
+            j.core.executors.run_local(f"{command} unplumb", hide=True)
+        if output.find("ipib") >= 0:
+            return "INFINIBAND"
+        else:
+            # work with interfaces which are subnetted on vlans eq e1000g5000:1
+            interfacepieces = interface.split(":")
+            interface = interfacepieces[0]
+            match = re.search(r"^\w+?(?P<interfaceid>\d+)$", interface, re.MULTILINE)
+            if not match:
+                raise j.exceptions.Value(f"Invalid interface {interface}")
+            if len(match.group("interfaceid")) >= 4:
+                return "VLAN"
+            else:
+                if len(interfacepieces) > 1:
+                    return "VIRTUAL"
+                else:
+                    return "ETHERNET_GB"
+
+
+def get_reachable_ip_address(ip: str, port: int):
```

Comment:
you can add `-> str` in the function signature to indicate the return type of the function